lib/app_profiler/backend/vernier_backend.rb

```diff
@@ -74,11 +76,19 @@ def results
 
         return unless vernier_profile
 
+        # Store all vernier metadata
+        meta = vernier_profile.meta.reject { |k, v| k == :user_metadata || v.nil? }
+        meta.merge!(@metadata) if @metadata
+
+        # Internal metadata takes precedence over user metadata, but store
+        # everything in user metadata
+        vernier_profile.meta[:user_metadata]&.merge!(meta)
+
         # HACK: - "data" is private, but we want to avoid serializing to JSON then
         # parsing back from JSON by just directly getting the hash
         data = ::Vernier::Output::Firefox.new(vernier_profile).send(:data)
         data[:meta][:mode] = @mode # TODO: https://github.com/jhawthorn/vernier/issues/30
-        data[:meta].merge!(@metadata) if @metadata
+        data[:meta][:vernierUserMetadata] ||= meta # for compatibility with < 1.7.0
```

Comment:
for metadata, we now read directly from this output key. To be compatible with older versions, we'll store the metadata that was added by the user here